contracts/RelaySet.sol

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018 Parity Technologies Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A validator set contract that relays calls to an inner validator set
+// contract, which allows upgrading the inner validator set contract. It
+// provides an `initiateChange` function that allows the inner contract to
+// trigger a change, since the engine will be listening for events emitted by
+// the outer contract. It keeps track of finality of pending changes in order to
+// validate `initiateChange` and `finalizeChange` requests.
+
+pragma solidity ^0.4.22;
+
+import "./Owned.sol";
+import "./ValidatorSet.sol";
+
+
+contract OuterSet is Owned, ValidatorSet {
+	// EVENTS
+	event ChangeFinalized(address[] currentSet);
+
+	// STATE
+
+	// System address, used by the block sealer.
+	address public systemAddress;
+	// Address of the inner validator set contract
+	InnerSet public innerSet;
```

Comment:
Maybe let's avoid generic types here and rather use `InnerOwnedSet` directly without any interface.